src/python/pants/option/arg_splitter_test.py

```diff
@@ -54,37 +54,60 @@ def assert_valid_split(
     splitter = ArgSplitter(ArgSplitterTest._known_scope_infos)
     args = shlex.split(args_str)
     split_args = splitter.split_args(args)
-    self.assertEqual(expected_goals, split_args.goals)
-    self.assertEqual(expected_scope_to_flags, split_args.scope_to_flags)
-    self.assertEqual(expected_specs, split_args.specs)
-    self.assertEqual(expected_passthru, split_args.passthru)
-    self.assertEqual(expected_passthru_owner, split_args.passthru_owner)
-    self.assertEqual(expected_is_help, splitter.help_request is not None)
-    self.assertEqual(expected_help_advanced,
-                      (isinstance(splitter.help_request, OptionsHelp) and
-                       splitter.help_request.advanced))
-    self.assertEqual(expected_help_all,
-                      (isinstance(splitter.help_request, OptionsHelp) and
-                       splitter.help_request.all_scopes))
-    self.assertEqual(expected_unknown_scopes, split_args.unknown_scopes)
+    assert expected_goals == split_args.goals
+    assert expected_scope_to_flags == split_args.scope_to_flags
+    assert expected_specs == split_args.specs
+    assert expected_passthru == split_args.passthru
+    assert expected_passthru_owner == split_args.passthru_owner
+    assert expected_is_help == (splitter.help_request is not None)
+    assert expected_help_advanced == (
+      isinstance(splitter.help_request, OptionsHelp) and splitter.help_request.advanced
+    )
+    assert expected_help_all == (
+      isinstance(splitter.help_request, OptionsHelp) and splitter.help_request.all_scopes
+    )
+    assert expected_unknown_scopes == split_args.unknown_scopes
+
+  def assert_unknown_goal(self, args_str: str, unknown_goals: List[str]) -> None:
+    splitter = ArgSplitter(ArgSplitterTest._known_scope_infos)
+    result = splitter.split_args(shlex.split(args_str))
+    assert isinstance(splitter.help_request, UnknownGoalHelp)
+    assert set(unknown_goals) == set(splitter.help_request.unknown_goals)
+    assert result.unknown_scopes == unknown_goals
```

Comment:
This was previously inlined, but it's now useful in two different tests so I promoted it to a method.